lib/cdn-site-hosting/cdn-site-hosting-construct.ts

```diff
@@ -67,36 +61,77 @@ export class CdnSiteHostingConstruct extends cdk.Construct {
     });
     new cdk.CfnOutput(this, "Bucket", { value: this.s3Bucket.bucketName });
 
+    const defaultSecurityHeaders: cloudfront.ResponseSecurityHeadersBehavior = {
+      // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src
+      contentSecurityPolicy: {
+        contentSecurityPolicy: "default-src 'self';",
+        override: true,
+      },
+      // https://web.dev/security-headers/#xcto
+      contentTypeOptions: { override: true },
+      // https://web.dev/security-headers/#recommended-usages-4
+      frameOptions: {
+        frameOption: cloudfront.HeadersFrameOption.DENY,
+        override: true,
+      },
+      // https://web.dev/referrer-best-practices/#setting-your-referrer-policy:-best-practices
+      referrerPolicy: {
+        referrerPolicy:
+          cloudfront.HeadersReferrerPolicy.STRICT_ORIGIN_WHEN_CROSS_ORIGIN,
+        override: true,
+      },
+      // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security#examples
+      strictTransportSecurity: {
+        accessControlMaxAge: Duration.millis(31536000),
+        includeSubdomains: true,
+        preload: true,
+        override: true,
+      },
+      // xxs-protection is overridden by the contentSecurityPolicy in modern browsers
+      // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection
+      xssProtection: { protection: false, override: true },
+    };
+
+    const responseHeadersPolicy = new cloudfront.ResponseHeadersPolicy(
+      this,
+      "ResponseHeadersPolicy",
+      {
+        responseHeadersPolicyName: "DefaultPolicy",
+        comment: "A default policy",
+        securityHeadersBehavior: {
+          ...defaultSecurityHeaders,
+          ...props.securityHeaders,
+        },
+      }
+    );
+
     // Cloudfront distribution
-    this.cloudfrontWebDistribution = new cloudfront.CloudFrontWebDistribution(
+    this.cloudfrontDistribution = new cloudfront.Distribution(
       this,
       "SiteDistribution",
       {
-        viewerCertificate,
-        originConfigs: [
-          {
-            // We use a custom origin rather than S3 origin because the latter
-            // does not seem to support websiteErrorDocument correctly
```

Comment:
question: Is this no longer an issue?

Reply:
There is no custom origin support in the new class as far as I am aware, I tried going to a bogus page in Otis in a stack I bought up and it returned the styled Otis error page rather than something from s3 so assume it works in the new class?

Reply:
I think the `customOriginSource` was a hold-your-nose solution because the default documented solution for attaching to an S3 bucket didn't seem to work via CDK.

This may be another benefit of the change, rather than something we need to recreate.